UPGRADE-18.0.md

```diff
@@ -584,3 +584,22 @@ resource "aws_iam_role_policy_attachment" "default" {
   policy_arn = aws_iam_policy.default.arn
 }
 ```
+### Reusing existing Cluster IAM Role
+
+If you allowed the module to create your Cluster IAM Role
+and want to reuse it moving forward you must
+migrate it into the state.
```

Comment:
```suggestion
When upgrading from v17.x to v18.x using the previously generated cluster IAM role, you will need to make a few changes to avoid replacing/re-creating the role which will cause the cluster to be replaced.

First, you will need to perform a Terraform state mv to "move" the previous resource name into the new resource name. See state move command below:

```bash
terraform state mv 'module.eks.aws_iam_role.cluster[0]' 'module.eks.aws_iam_role.this[0]'
```


Next, you will need to configure a few settings that will align your role with the v17.x version and avoid replacing/re-creating. See the associated settings below that should be set:

```hcl
  iam_role_name            = <name of current role created by v17.x module>
	iam_role_use_name_prefix = false
```
```

We don't need to tell users to specify all those settings because they do not make sense. If you set the prefix separator to an empty string, but then follow it up with `iam_role_use_name_prefix = false`, it doesn't make sense because we've ignored its use.

Also, we don't need to specify the role ARN because that is only for externally managed roles.